### PR TITLE
feat(FEC-8046): 360 support

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

* overlay-action:
     * pass `isVr` prop
     * handle click by `pointerdown`/`pointerup` events instead of bu `click` to distinguish between 
       click and dragging. 
* change the shell touch handler as a result the overlay-action changes 
* add stereo control 
* trigger `window.resize` on in-browser fullscreen actions

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/middlewares/event-dispatcher.js [hunk 1]
```diff
@@ -135,6 +135,10 @@ function onClickableComponentsHandler(store: any, action: Object, player: Player
     case 'Volume':
       onVolumeClicked(store, action, player);
       break;
+
+    case 'Stereo':
+      player.dispatchEvent(new FakeEvent(FakeEvent.Type.USER_CLICKED_STEREO_MODE));
```
Comment (dan-ziv) [inline src/middlewares/event-dispatcher.js]: Maybe add a payload which indicates whether the stereo is on or off?

**Comment by dan-ziv** [comment]
@yairans 
add the new UI event to docs:
https://github.com/kaltura/playkit-js-ui/blob/master/docs/events.md
+
The new config for vr component if such exists

File: src/components/stereo/stereo.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  /**
+   * Stereo click handler
+   *
+   * @returns {void}
+   * @memberof StereoControl
+   */
+  onClick(): void {
+    this.notifyClick();
```
Comment (OrenMe) [inline src/components/stereo/stereo.js]: see https://github.com/kaltura/kaltura-player-js/pull/130/files#r196646166 and https://github.com/kaltura/playkit-js/pull/239/files#r196647201 use `player.toggleVrStereoMode()`
Also need to be reactive to VR events, so need to add a `vrStereoModeChanged` event (when change is done via API of player) and listen to it and change state accordingly.

File: src/event/event-type.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ const EventType: { [event: string]: string } = {
   USER_CLICKED_LIVE_TAG: `${namespace}-userclickedlivetag`,
   USER_CLICKED_MUTE: `${namespace}-userclickedmute`,
   USER_CLICKED_UNMUTE: `${namespace}-userclickedunmute`,
+  USER_CLICKED_STEREO_MODE: `${namespace}-userclickedstereomode`,
```
Comment (OrenMe) [inline src/event/event-type.js]: `userclickedvrstereomode`

File: src/middlewares/event-dispatcher.js (hunk 1 above)
Comment (OrenMe) [reply src/middlewares/event-dispatcher.js]: second @dan-ziv 's comment, also change to  `USER_CLICKED_VR_STEREO_MODE`

File: src/reducers/config.js [hunk 4]
```diff
@@ -10,7 +10,8 @@ export const initialState = {
   components: {
     watermark: {},
     seekbar: {},
-    error: {}
+    error: {},
+    stereo: {},
```
Comment (OrenMe) [inline src/reducers/config.js]: `vrStereoToggle`

File: src/reducers/stereo.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+//@flow
+
+import {getComponentStateFromComponentConfig, getComponentStateFromConfig} from "../utils/component-config";
```
Comment (OrenMe) [inline src/reducers/stereo.js]: align to naming convention as well `vrStereoToggle`

File: src/reducers/stereo.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    case configReducerTypes.UPDATE:
+      return getComponentStateFromConfig(component, state, action);
+
+    case configReducerTypes.UPDATE_COMPONENT:
+      return getComponentStateFromComponentConfig(component, state, action);
+
+    case types.UPDATE_STEREO_MODE:
+      return {
+        ...state,
+        stereoMode: action.stereoMode
```
Comment (OrenMe) [inline src/reducers/stereo.js]: `vrStereoMode`

File: src/components/overlay-action/overlay-action.js [hunk 7]
```diff
@@ -115,6 +118,60 @@ class OverlayAction extends BaseComponent {
...
+  }
+
+  /**
+   * Whether the user is dragging
+   *
+   * @param {*} event - mouseup/touchend event
+   * @returns {boolean} - is dragging
+   */
+  isDragging(event: any): boolean {
+    if (event.clientX) {
```
Comment (OrenMe) [inline src/components/overlay-action/overlay-action.js]: you can normalize the event to:
```javascript
const points = {
   clientX: event.clientX || event.changedTouches[0] && event.changedTouches[0].clientX,
   clientY: event.clientY || event.changedTouches[0] && event.changedTouches[0].clientY
}
return (points.clientX !== this._pointerDownPosX || points.clientY !== this._pointerDownPosY);
```

Comment (yairans) [reply src/components/overlay-action/overlay-action.js]: @OrenMe  `event.changedTouches[0] && event.changedTouches[0].clientX` is not a boolean?

**Pull request merged**